File: javascript/algorithms/insertionSort.js
const insertionSort = (values) => {
  for (let i = 0; i < values.length; i++) {
    let value = values[i];
    for (let j = i - 1; j > -1 && values[j] > value; j--) {
      values[j + 1] = values[j];
    }

    values[j + 1] = value;
  }

  return values;
}

console.log(insertionSort([99, 238, 0, 11, -2, 1, 19, 0, 11]));

File: swift/algorithms/InsertionSort.swift


import Foundation


func InsertionSort(_ a:inout [Int]){
    
    for i in 1..<a.count{
        
        let temp = a[i]
        var k = i-1
        while k>=0 && a[k]>temp{
            a[k+1]=a[k]
            k = k-1
            
            }
        
        a[k+1] = temp
        }
    
    }


//Testing code

var a = [100,58,0,9,22,3,21,23,4,1,3]
InsertionSort(&a)
print(a)
